kmip/pie/client.py

```diff
@@ -0,0 +1,380 @@
+# Copyright (c) 2015 The Johns Hopkins University/Applied Physics Laboratory
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import logging
+import multiprocessing
+import six
+import ssl
+
+from kmip.core import enums
+from kmip.core import objects as cobjects
+
+from kmip.core.factories import attributes
+
+from kmip.pie import exceptions
+from kmip.pie import factory
+from kmip.pie import objects as pobjects
+
+from kmip.services.kmip_client import KMIPProxy
+
+
+class KmipClient:
+    """
+    A simplified PyKMIP client for conducting KMIP operations.
+
+    The KmipClient provides a simpler external interface for various KMIP
+    operations and composes the bulk of the PyKMIP Pie API. It wraps the
+    original KMIPProxy, reducing the boilerplate needed to deploy PyKMIP
+    in client applications.
+    """
+    def __init__(self,
+                 hostname='127.0.0.1',
+                 port=5696,
+                 cert=None,
+                 key=None,
+                 ca=None,
+                 ssl_version=ssl.PROTOCOL_SSLv23,
```

Comment:
After testing this operationally, I'm actually going to remove all default values from the client constructor, aside from the config value. If a default is specified in the code, it will override the config file value which I've found in testing to be mildly unintuitive. Since a user may not see the default value in the client constructor call, but will likely see the value in the config file, if they differ and the client constructor default is used, it's not clear why the config file value isn't used. We can either update the basic config to use TLS 1.0 or change the way we dynamically load config values to make this more intuitive to users.

For now, I'm just making them all `None`.